FreeRTOS_ARP.c

```diff
@@ -61,10 +61,19 @@
     #define arpGRATUITOUS_ARP_PERIOD    ( pdMS_TO_TICKS( 20000U ) )
 #endif
 
+/** @brief When there is another device which has the same IP address as the IP address
+ * of this device, a defensive ARP request should be sent out. However, according to
+ * RFC 5227 section 1.1, there must be a minimum interval of 10 seconds between
+ * consecutive defensive ARP packets. */
+#define arpIP_CLASH_RESET_TIMEOUT    10000
+
+/** @brief Maximum number of defensive ARPs to be sent for an ARP clash per
+ * arpIP_CLASH_RESET_TIMEOUT period. */
+#define arpIP_CLASH_MAX_RETRIES      1
```

Comment:
Should both of these macros be wrapped in `#ifndef`..`#endif` so that they can be configured? Why is 1 the best default?

Reply:
Indeed, that would be nice.
As you may have seen, there are two types of macros:
- E.g. `ipconfigSOCK_DEFAULT_RECEIVE_BLOCK_TIME` which is a global macro, it gets a default value in FreeRTOSIPConfigDefaults.h.
- E.g. `arpGRATUITOUS_ARP_PERIOD`, which is local to FreeRTOS_ARP.c. it is more an expert option but can be overwritten as well.

So I also vote for `#ifndef` `#endif`.

@yourslab wrote:
> Why is 1 the best default?

That used to be the default, I think?

Reply:
I agree that we should wrap these in `#ifndef`...`#endif`.

1 is the values suggested by `RFC 5227 section 2.4 part (b)` [here](https://datatracker.ietf.org/doc/html/rfc5227#section-2.4). I shall update my comment to reflect why was this value chosen.

It also suggests that if there are more conflicts in the DEFEND_INTERVAL, we should cease to use the IP address immediately - however, that is quite complex to do - considering we might have ongoing TCP connections and we don't support ICMP packets except request and response.